.github/workflows/create-stable-release.yml

```diff
@@ -0,0 +1,116 @@
+# Automatically create new releases from the main branch when there are new commits since the last stable release.
+name: Create stable release
+
+on:
+  workflow_call:
+    secrets:
+      token:
+        required: true
+        description: GitHub token with permissions to create releases. Prefer personal access tokens over default GitHub token, so tags created by this template can trigger other workflows.
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Required to create a release
+
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - name: Get latest stable tag
```

Comment:
I wonder if it would not be easier to have a single step instead of many `$env:GITHUB_OUTPUT` which may be harder to follow.

Note that for logs, you can always create manual groups if needed: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#grouping-log-lines

Reply:
I had a similar thought. While coding this, I ended up liking this separation of steps because it gave me a better idea of where things went wrong.

Reply:
I also like seeing the steps being skipped when there is no commits

![image](https://github.com/user-attachments/assets/3348217e-8421-40a1-a953-ed18b3e07983)